File: test-suite/js/suppression/suppression_buggy_nomarkers.js
// Suppression fixture pair (bead A7, GH #91) for the JS/TS module.
//
// Twin of suppression_buggy.js: every line is aligned with the markered
// fixture but the suppression markers are stripped, so the findings that
// the markered file suppresses must be reported here. The four marker
// arrangements exercised are:
//   1. trailing the flagged line,
//   2. on the line immediately above the flagged statement,
//   3. on a middle physical line of a multi-line statement,
//   4. formatter-relocated onto the first line inside a block that the
//      flagged statement opens.
// A scan of this file must report at least two findings.

// ---------------------------------------------------------------------------
// Arrangement 1 -- trailing marker: the dynamic-code evaluation call is a
// critical finding for the heuristic pass and for ast-grep rule js.eval-call.
// ---------------------------------------------------------------------------
function runUserCode(userCode) {
  eval(userCode); // trailing marker arrangement (marker stripped here)
}

// ---------------------------------------------------------------------------
// Arrangement 2 -- previous-line marker: a literal assigned to a
// secret-named constant is a critical hardcoded-secret finding.
// ---------------------------------------------------------------------------
// previous-line marker arrangement (marker stripped here)
const API_SECRET = "sk-live-suppression-fixture-0001";

// ---------------------------------------------------------------------------
// Arrangement 3 -- multi-line statement middle: the token comparison sits on
// the middle physical line of a multi-line `if (...)` condition.
// ---------------------------------------------------------------------------
function compareTokens(authToken, expectedToken) {
  if (
    authToken !== expectedToken // multi-line middle marker (stripped here)
  ) {
    return false;
  }
  return true;
}

// ---------------------------------------------------------------------------
// Arrangement 4 -- formatter-relocated marker: the flagged `if (...) {`
// opens a block and the marker moved to its first line, as formatters do.
// ---------------------------------------------------------------------------
function checkSession(sessionToken, expectedSessionToken) {
  if (sessionToken !== expectedSessionToken) {
    // formatter-relocated marker arrangement (marker stripped here)
    return false;
  }
  return true;
}

File: test-suite/golang/suppression/suppression_buggy_nomarkers.go
// GH #91 regression fixtures: golang suppression divergence pair (bead A7).
// Each function here triggers a native ubs-golang.sh finding that the sibling
// suppression_buggy.go suppresses with an inline ignore marker in a distinct
// placement: trailing, previous-line, multi-line statement,
// formatter-relocated first line inside a block, and rule-scoped.
// This twin carries no markers, so every finding must be reported.

package suppression

import (
	"fmt"
	mathrand "math/rand"
	"net/http"
)

// Placement 1 (trailing): the marker rides the flagged declaration itself.
func trailingMarkerAssignment() string {
	apiKey := "ak_live_9f2c7b61d4e8a5c3"
	return apiKey
}

// Placement 2 (previous-line): the marker sits on the line directly above the
// flagged redirect sink.
func prevLineRedirect(w http.ResponseWriter, r *http.Request) {
	http.Redirect(w, r, r.Header.Get("X-Next-URL"), http.StatusFound)
}

// Placement 3 (multi-line statement): the flagged assignment opens a
// statement that continues onto the next physical line, so the trailing
// marker added in the twin sits on the opening line of a multi-line
// statement.
func multilineStatementAssignment() string {
	apiKeyExtended := "ak_live_9f2c7b61d4e8a5c3" +
		"-extended"
	return apiKeyExtended
}

// Placement 4 (formatter-relocated): formatters push a trailing marker off
// the block-opening line to the first line inside the block.
func relocatedCookie(w http.ResponseWriter, r *http.Request) {
	sid := r.Header.Get("X-Session-Id")
	if sid != "" {
		http.SetCookie(w, &http.Cookie{Name: "sid", Value: sid, Secure: false, HttpOnly: false})
	}
}

// Placement 5 (rule-scoped): the marker names the single rule it suppresses.
func scopedRandomToken() string {
	token := fmt.Sprintf("%d", mathrand.Int63())
	return token
}
